EXILED/Exiled.Events/Patches/Events/Scp330/InteractingScp330.cs

```diff
@@ -73,29 +75,46 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                     //    return;
                     new(OpCodes.Callvirt, PropertyGetter(typeof(InteractingScp330EventArgs), nameof(InteractingScp330EventArgs.IsAllowed))),
                     new(OpCodes.Brfalse, returnLabel),
+
+                    // ldarg.1 to player
+                    new(OpCodes.Ldarg_1),
+
+                    // ev.Candy
+                    new(OpCodes.Ldloc, ev.LocalIndex),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(InteractingScp330EventArgs), nameof(InteractingScp330EventArgs.Candy))),
+
+                    // bag
+                    new(OpCodes.Ldloca_S, 3),
+
+                    // ServerProcessPickup(ReferenceHub, CandyKindID, Scp330Bag)
+                    new CodeInstruction(OpCodes.Call, Method(typeof(InteractingScp330), nameof(ServerProcessPickup), new[] { typeof(ReferenceHub), typeof(CandyKindID), typeof(Scp330Bag).MakeByRefType() })),
                 });
 
+            int remove_offset = -3;
+            int remove_index = newInstructions.FindLastIndex(
+                instruction => instruction.Calls(Method(typeof(Scp330Bag), nameof(Scp330Bag.ServerProcessPickup)))) + remove_offset;
+
+            // Remove original add candy logic
+            newInstructions.RemoveRange(remove_index, 4);
+
             /* next code will used to override sound rpc check by EXILED
              * old:
              *   if (args.PlaySound)
              * new:
-             *   if (args.PlaySound && ev.PlaySound)
+             *   if (args.PlaySound || ev.PlaySound)
```

Comment:
actually the code look like this
```suggestion
             *   if (args.PlaySound | ev.PlaySound)
```

Reply:
Ya got me there